src/components/share/share.js

```diff
@@ -40,6 +49,17 @@ class ShareControl extends BaseComponent {
    */
   toggleOverlay(): void {
     this.setState({overlay: !this.state.overlay});
+    this.props.toggleShareOverlay(this.state.overlay);
+    if (this.state.overlay) {
```

Comment:
need to keep local player state and only resume if it was playing before share was clicked.

Reply:
this logic should be applied in any overlay (advanced captions settings for example) no?